cloudinit/config/cloud-init-schema.json

```diff
@@ -1343,6 +1343,303 @@
+          "patternProperties": {
+            "^(rsa|dsa|ecdsa|ed25519)_(public|private|certificate)$": {
+              "label": "<key_type>",
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        "ssh_authorized_keys": {
+          "type": "array",
+          "minItems": 1,
+          "description": "The SSH public keys to add ``.ssh/authorized_keys`` in user's home directory",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ssh_deletekeys" : {
+          "type": "boolean",
+          "default": true,
+          "description": "Remove host SSH keys. This prevents re-use of a private host key from an image with default host SSH keys. Default: ``true``"
+        },
+        "ssh_genkeytypes": {
+          "type": "array",
+          "description": "The SSH key types to generate. Default: ``[rsa, dsa, ecdsa, ed25519]``",
+          "default": ["rsa", "dsa", "ecdsa", "ed25519"],
+          "minItems": 1,
+          "items": {
+            "type": "string",
+            "enum": ["rsa", "dsa", "ecdsa", "ed25519"]
+          }
+        },
+        "disable_root": {
+          "type": "boolean",
+          "default": true,
+          "description": "Disable root login. Default: ``true``"
+        },
+        "disable_root_opts": {
+          "type": "string",
+          "description": "Disable root login options.  If ``disable_root_opts`` is specified and contains the string ``$USER``, it will be replaced with the username of the default user. Default: ``no-port-forwarding,no-agent-forwarding,no-X11-forwarding,command=\"echo 'Please login as the user \\\"$USER\\\" rather than the user \\\"$DISABLE_USER\\\".';echo;sleep 10;exit 142\"``"
+        },
+        "allow_public_ssh_keys": {
+          "type": "boolean",
+          "default": true,
+          "description": "If ``true``, will import the public SSH keys from the datasource's metadata to the user's ``.ssh/authorized_keys`` file. Default: ``true``"
+        },
+        "ssh_quiet_keygen": {
+          "type": "boolean",
+          "default": false,
+          "description": "If ``true``, will suppress the output of key generation to the console. Default: ``false``"
+        },
+        "ssh_publish_hostkeys": {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "default": true,
+              "description": "If true, will read host keys from /etc/ssh/*.pub and publish them to the datasource (if supported). Default: ``true``"
+            },
+            "blacklist": {
+              "type": "array",
+              "description": "The SSH key types to blacklist when publishing. Default: ``[dsa]``",
```

Comment:
```suggestion
              "description": "The SSH key types to ignore when publishing. Default: ``[dsa]``",
```

Reply:
Once we add the ability to hide schema properties from the docs, I'd like to deprecate this and replace it with 'denylist'